File: 0169-majority-element/0169-majority-element.py

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        ele=None
        cnt=0
        for i in range(len(nums)):
            if cnt==0:
                ele=nums[i]
                cnt=1
            elif nums[i]==ele:
                cnt+=1
            else:
                cnt-=1
        
        cnt1=0
        for i in range(len(nums)):
            if nums[i]==ele:
                cnt1+=1
        
        if cnt1>=len(nums)/2:
            return ele
        
        return -1

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int ele=0;
        int count=0;
        for(int num:nums){
            if(count==0){
                ele=num;
                count++;
            }else if(num==ele){
                count++;
            }else{
                count--;
            }
        }
        int maxCount=0;
        for(int num:nums){
            if(num==ele){
                maxCount++;
            }
        }
        int n=nums.length;
        if(maxCount>=n/2) return ele;
        return -1;
    }
}